spec/09-lexical-structure.md

```diff
@@ -678,7 +678,7 @@ The variable substitution accepts the following syntax:
 * [*name*](#names)
 * [*variable-name*](#names)
 
-*expression* works the same way as in [variable name creation operator](10-expressions.md#variable-name-creation-operator).
+The `${ expression }` syntax is deprecated in Hack. The expression must be such that if the enclosing braces were removed and the `$` prepended, then the resulting expression would be a legal <i>`string-variable`</i>. For example, `"${y[123]}"` is legal because `$y[123]` is a legal <i>`string-variable`</i>. No whitespace or comments are allowed inside the braces.
```

Comment:
Great feedback, thanks. 

Couple things.

First, I totally agree about legal vs valid. I'll make those fixes.

Second, I did not realize when I wrote this that the specification says in section 4 that the specification is only specifying "strict" mode. What I intended to convey by the comment about deprecation is that this syntactic form produces an error in strict mode. 

So I'm not sure what to do at this point. We could (1) remove it entirely and be done with it, or (2) add a note here saying that this syntax is permitted in non-strict mode, or (3) remove it from here, make a new appendix that is "common deviations from strict mode", (4) leave the text as-is and not stress about it, or (5) do something I haven't thought of.  Thoughts and opinions?